client/src/main/java/org/apache/falcon/entity/v0/EntityType.java

```diff
@@ -124,4 +128,27 @@ public static EntityType getEnum(String type) {
                     + Arrays.toString(values()).toLowerCase() + ".");
         }
     }
+
+    /**
+     * Finds the value of the given enumeration by name, case-insensitive.
+     * Throws an IllegalArgumentException if no match is found.
+     *
+     * @param entityType : string to compare against the value of EntityType
+     * @return EntityType : Object of type EntityType
+     * @throws IllegalArgumentException incase the argument is not present in the EntityType enum
+     **/
+
+    public static EntityType fromString(String entityType) {
```

Comment:
There is already a getEnum method in EntityType, which converts a given string to uppercase and then validates.

As discussed, ideally, all the internal classes understand EntityType enum, if the DB requires it as a string, the Bean should handle internal conversion for read/write appropriately.